src/server/common/model/state/state-manager.js

```diff
@@ -4,9 +4,11 @@
 
 /**
  * @typedef {Record<string, any>} RawSectionState
- * @typedef {Record<string, RawSectionState>} RawApplicationState
+ * @typedef {{ application: RawSectionState}} RawApplicationState
+ * @typedef  {'origin' |  'destination' | 'licence' | 'identification' | 'biosecurity' | 'biosecurity-map'} SectionKeys
```

Comment:
As per our conversation elsewhere, I think this is where the complexity in the refactor has creeped in. 

Originally, RawApplicationState looks like (for instance)

```javascript
{
  origin: { 
    onOffFarm: 'on',
    address: { ... }
  },
  destination: { ... }
}
```

If we keep this interface the same, then the refactoring surface gets a lot smaller.  (Also, I think the structure is neater, because we avoid one level of nesting which isn't serving a strong purpose)